### PR TITLE
Batch fetch receipts in EthConfirmer

### PR DESCRIPTION
**Pull request by samsondav**
This ought to hopefully improve performance when we have a lot of
transactions in-flight.

### CODE REVIEW

File: core/services/bulletprooftxmanager/eth_confirmer.go [hunk 1]
```diff
@@ -140,126 +157,135 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
...
+		ON CONFLICT (tx_hash, block_hash) DO UPDATE SET
+			block_number = EXCLUDED.block_number,
+			transaction_index = EXCLUDED.transaction_index,
+			receipt = EXCLUDED.receipt
+		RETURNING eth_receipts.tx_hash, eth_receipts.block_number
+	),
+	updated_eth_tx_attempts AS (
+		UPDATE eth_tx_attempts
+		SET
+			state = 'broadcast',
```
Comment (connorwstein) [inline core/services/bulletprooftxmanager/eth_confirmer.go:274]: nit - could we use the typed strings here has args to the sprintf

File: core/services/eth/client.go [hunk 2]
```diff
@@ -362,3 +363,10 @@ func (client *client) CallContext(ctx context.Context, result interface{}, metho
 	)
 	return client.RPCClient.CallContext(ctx, result, method, args...)
 }
+
+func (client *client) BatchCallContext(ctx context.Context, b []rpc.BatchElem) error {
+	logger.Debugw("eth.Client#BatchCall(...)",
+		"batchElems", b,
+	)
+	return client.RPCClient.BatchCallContext(ctx, b)
```
Comment (connorwstein) [inline core/services/eth/client.go:371]: this gets tested against geth and parity in the integration tests right?

**Review by connorwstein — APPROVED** [review]

**Comment by connorwstein** [comment]
nice work man!

File: core/services/bulletprooftxmanager/eth_confirmer.go (hunk 1 above)
Comment (samsondav) [reply core/services/bulletprooftxmanager/eth_confirmer.go:274]: Why?

File: core/services/eth/client.go (hunk 2 above)
Comment (samsondav) [reply core/services/eth/client.go:371]: Yes I believe so

File: core/services/bulletprooftxmanager/eth_confirmer.go (hunk 1 above)
Comment (connorwstein) [reply core/services/bulletprooftxmanager/eth_confirmer.go:274]: oh didn't know these were enums in the db (thought they were strings), makes sense then

**Review by connorwstein — APPROVED** [review]

**Pull request merged**